src/cluster/replication.cc

```diff
@@ -203,6 +206,21 @@ void FeedSlaveThread::loop() {
         Stop();
         return;
       }
+
+      // Check if this change would unblock any WAIT command
+      auto largest_unblockable_seq = srv_->LargestTargetSeqToWakeup(batch.sequence);
+      if (largest_unblockable_seq > last_replconf_getack_seq_) {
+        // Send _getack to the slave to get acknowledgment
+        auto s = util::SockSend(conn_->GetFD(), redis::BulkString("_getack"), conn_->GetBufferEvent());
+        if (!s.IsOK()) {
+          error("Write error while sending _getack to slave: {}", s.Msg());
+          Stop();
+          return;
+        } else {
+          last_replconf_getack_seq_ = largest_unblockable_seq;
```

Comment:
Should we set `last_replconf_getack_seq_` to `largest_unblockable_seq` is zero? For example, when the wait list is empty.

Reply:
```
Should we set last_replconf_getack_seq_ to largest_unblockable_seq when the value of largest_unblockable_seq is zero?
```
I am not sure if I get you.

`last_replconf_getack_seq_` is used to prevent unnecessary `_getack` sent. If we set it back to zero, it would cause unnecessary sends.

The check in `largest_unblockable_seq > last_replconf_getack_seq_` ensures we would not set `last_replconf_getack_seq_` back to zero when `largest_unblockable_seq` is zero.

Reply:
I renamed `last_replconf_getack_seq_` to `last_getack_seq` and make it a local var.

Reply:
good call to reduce `util::SockSend` call, let me update it

Reply:
> The check in largest_unblockable_seq > last_replconf_getack_seq_ ensures we would not set last_replconf_getack_seq_ back to zero when largest_unblockable_seq is zero.

Yes, exactly. Thanks for your response and update.